src/components/Topic.jsx

```diff
@@ -0,0 +1,16 @@
+import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { getArticles } from '../utils/apis'
+import ArticlePreview from './ArticlePreview'
+import ResultsList from './ResultsList'
+
+const Topic = ({ allTopics }) => {
+    const params = useParams()
+
+    return <section className="">
+        <h3 className="articles-header">{params.topic} Articles</h3>
+        < ResultsList allTopics={allTopics} topic={params.topic} />
+    </section>
+}
+
+export default Topic
```

Comment:
This Topic component seems very similar to the Home component, (both leading to the ResultsPage component) - other than the fact the Topic component uses params. There could be a case for refactoring them into one component that acts differently depending on the params object.
(this isn't mandatory though just a suggestion)